### ISSUE TITLE
Deal with multiple outputs

### ISSUE THREAD

**Issue by texttheater** [texttheater/produce#7]
Compare the following Producefile and Makefile, which should be roughly equivalent:

```
[]
default = a.txt b.txt

[%{name}.txt]
cond = %{name in ('a', 'b')}
recipe = touch a.txt b.txt
```

```
default : a.txt b.txt

a.txt b.txt :
        touch a.txt b.txt
```

But Produce and Make behave differently:

```
$ rm -f a.txt b.txt
$ produce
INFO: building file a.txt
    touch a.txt b.txt
INFO: rebuilding file b.txt
    touch a.txt b.txt
$ rm a.txt b.txt
$ make
touch a.txt b.txt
$
```

On dealing with the second target (b.txt), Make notices that it was already updated, Produce does not. This is because Produce traverses the dependency tree twice, first checking which targets are out of date, then running the required recipes. By the time the first recipe runs, whether to run the second one has already be determined.

Can we instead do it all in one pass? This would also make dealing with dependency files more pleasant and probably simplify the code overall.

But I'm sure there was a good reason for two passes. Can't seem to remember ATM.

**Comment by texttheater** [comment]
Make is however also limited in this respect. If we want this to work properly in parallel builds, we must introduce a way to explicitly declare multiple outputs: http://blog.melski.net/2013/01/01/pragma-multi-and-rules-with-multiple-outputs-in-gnu-make/

**Comment by texttheater** [comment]
In place in branch feat/outputs, no regression tests yet.

**Comment by texttheater** [comment]
Fixed in c426fa281b2d405fbd710012921988bfcfba0b53.
